frontend/src/components/HardwareDialog.tsx

```diff
@@ -0,0 +1,78 @@
+import { useState, useRef, useEffect } from "react";
+
+interface HardwareDialogProps {
+    isOpen: boolean; // Determines if the dialog is open
+    onClose: () => void; // Function to close the dialog
+    onSubmit: (action: "checkOut" | "checkIn", hardwareId: string, quantity: number) => void; // Function to handle submit
+}
+
+//Component Setup and State Inits
+const HardwareDialog = ({ isOpen, onClose, onSubmit }: HardwareDialogProps) => {
```

Comment:
Remember that this is a self-contained component, not a generic one like the user-form. So this shouldn't have any props, and should maintain all state internally and call the API itself